myia/anf_ir.py

```diff
@@ -157,7 +106,7 @@ def __init__(self, inputs: Iterable['ANFNode'], value: Any,
         self.value = value
         self.graph = graph
         self.uses: Set[Tuple[ANFNode, int]] = set()
-        self.debug = ANFNodeDebug(self)
+        self.debug = AutoNamedDebugInfo(type='graph')
```

Comment:
`'graph'`? Also, it's hardcoded here whereas it's dynamic for `Graph`

Reply:
Yeah, that's because I wanted to preserve the same string representation you had, but I didn't want `DebugInfo` to hold a pointer to the object (because holding a `DebugInfo` shouldn't keep the object alive, especially if they're going to be chained in the `about` field), so I just used a string instead.

This being said, thinking about it, the object could be held in a weak reference. I think that would be perfect.

Reply:
Yeah I just realized what you meant. That's obviously the wrong type, oops.